src/manifest.ts

```diff
@@ -144,12 +249,109 @@ export class Manifest {
     return validConfig && validManifest;
   }
 
+  private async runReleasers(
+    packages: PackageReleaseData[],
+    sha?: string
+  ): Promise<[Map<string, string>, PackageWithPRData[]]> {
+    const manifestUpdates: VersionsMap = new Map();
+    const openPRPackages: PackageWithPRData[] = [];
+    for (const pkg of packages) {
+      const {releaseType, ...options} = pkg.releaserOptions;
+      const releaserClass = factory.releasePRClass(releaseType);
+      const releasePR = new releaserClass({github: this.gh, ...options});
+      const pkgName = (await releasePR.getPackageName()).name;
+      this.checkpoint(
+        `Processing package: ${releaserClass.name}(${pkgName})`,
+        CheckpointType.Success
+      );
+      const openPROptions = await releasePR.getOpenPROptions(
+        pkg.commits,
+        pkg.lastVersion
+          ? {
+              name: `v${pkg.lastVersion}`,
+              sha: sha ?? 'beginning of time',
+              version: pkg.lastVersion,
+            }
+          : undefined
+      );
+      if (openPROptions) {
+        openPRPackages.push({name: releasePR.packageName, openPROptions});
+        manifestUpdates.set(pkg.path, openPROptions.version);
+      }
+    }
+    return [manifestUpdates, openPRPackages];
+  }
+
+  private async buildManifestPR(
+    manifestUpdates: Map<string, string>,
+    openPRPackages: PackageWithPRData[]
+  ): Promise<[string, Update[]]> {
+    let body = ':robot: I have created a release \\*beep\\* \\*boop\\*';
+    const updates: Update[] = [];
+    for (const openPRPackage of openPRPackages) {
+      body +=
+        '\n\n---\n' +
+        `${openPRPackage.name}: ${openPRPackage.openPROptions.version}\n` +
+        `${openPRPackage.openPROptions.changelogEntry}`;
+      updates.push(...openPRPackage.openPROptions.updates);
+    }
+    updates.push(
+      new ReleasePleaseManifest({
+        changelogEntry: '',
+        packageName: '',
+        path: this.manifestFileName,
+        version: '',
+        versions: manifestUpdates,
+        contents: await this.getManifest(),
+      })
+    );
+    body +=
+      '\n\nThis PR was generated with [Release Please]' +
+      `(https://github.com/googleapis/${RELEASE_PLEASE}). See [documentation]` +
+      `(https://github.com/googleapis/${RELEASE_PLEASE}#${RELEASE_PLEASE}).`;
+    return [body, updates];
+  }
+
   async pullRequest(): Promise<number | undefined> {
     const valid = await this.validate();
     if (!valid) {
       return;
     }
-    return 1;
+
+    const branchName = (await this.getBranchName()).toString();
+    const lastMergedPR = await this.gh.lastMergedPRByHeadBranch(branchName);
```

Comment:
what would happen if we accidentally missed on merge, one of our submodules wouldn't get released until another commit touched it?

Reply:
is your question

1. "what would happen if `lastMergedPR` here is `undefined`?" - in this case the releaser falls back to getting version info from `.release-please-manifest.json` at HEAD if it exists and if it doesn't then it falls back to each package releaser's `defaultInitialVersion` method. See [this test](https://github.com/googleapis/release-please/pull/800/files#diff-1c245f2069be81e5dae96c50d3f5bac65a1c14be82e61e7d49f9ac89587f3af6R506)
2. what if a submodule has no user facing commits since the last merged PR? in that case it doesn't show up at all in the release PR - only packages with user facing changes have updates in the PR. See [this test](https://github.com/googleapis/release-please/pull/800/files#diff-1c245f2069be81e5dae96c50d3f5bac65a1c14be82e61e7d49f9ac89587f3af6R386)

Reply:
It seems like we get the last merged PR? was wondering if we happened to miss running for one PR, what problems it would cause -- If I'm understanding the code correctly, it would self correct eventually?

Reply:
> happened to miss running for one PR

I assume you mean creating github releases? if we went two rounds of

`manifest-pr` -> creates Release PR -> Release PR merged

without running `manifest-release` (Tag/Release) in between: the way I'm building `manifest-release` / `Manifest.githubRelease` currently you'd miss out on those tags/releases. I'll add a note for future functionality to run `manifest-release` with a specific sha or PR number to create releases for.